### PR TITLE
[1.18] .github: cleanup nightly-tests, add placeholder changelog

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description
 
Minor cleanup to the 1.18 branch, test CI works

# Context
We just cut the v1.18.x branch. I want to confirm that developers can merge code safely to it.

## Testing steps
- This PR can pass CI and merge with 2 approvals

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/docs-gen.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
   push:
     branches:
     - 'main'
+    - 'v18.x'
```
Comment (jenshu) [inline .github/workflows/docs-gen.yaml:10]:
```suggestion
    - 'v1.18.x'
```

**Comment by sam-heilbron** [comment]
/kick https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=4c3d7321-fca0-417a-a37c-d4f5709c92b9

```
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   [38;5;9m[FAILED][0m in [It] - /workspace/gloo/test/e2e/local_ratelimit_test.go:161 [38;5;243m@ 12/02/24 23:28:17.819[0m
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   [1mSTEP:[0m TestContext.AfterEach: Cancelling test context [38;5;243m@ 12/02/24 23:28:17.819[0m
Step #1 - "run-tests": Step #8 - "run-e2e-tests":     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Step #1 - "run-tests": Step #8 - "run-e2e-tests":                                    Dload  Upload   Total   Spent    Left  Speed
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 127.0.0.1:20074...
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   * connect to 127.0.0.1 port 20074 failed: Connection refused
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   * Failed to connect to 127.0.0.1 port 20074 after 0 ms: Couldn't connect to server
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   * Closing connection 0
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   curl: (7) Failed to connect to 127.0.0.1 port 20074 after 0 ms: Couldn't connect to server
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   [38;5;243m<< Timeline[0m
Step #1 - "run-tests": Step #8 - "run-e2e-tests": 
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   [38;5;9m[FAILED] The connection be limited
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   Expected
Step #1 - "run-tests": Step #8 - "run-e2e-tests":       <*url.Error | 0xc0028979b0>: 
Step #1 - "run-tests": Step #8 - "run-e2e-tests":       Get "http://localhost:9155/": http: server closed idle connection
Step #1 - "run-tests": Step #8 - "run-e2e-tests":       {
Step #1 - "run-tests": Step #8 - "run-e2e-tests":           Op: "Get",
Step #1 - "run-tests": Step #8 - "run-e2e-tests":           URL: "http://localhost:9155/",
Step #1 - "run-tests": Step #8 - "run-e2e-tests":           Err: <*errors.errorString | 0xcb26e40>{
Step #1 - "run-tests": Step #8 - "run-e2e-tests":               s: "http: server closed idle connection",
Step #1 - "run-tests": Step #8 - "run-e2e-tests":           },
Step #1 - "run-tests": Step #8 - "run-e2e-tests":       }
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   to match error
Step #1 - "run-tests": Step #8 - "run-e2e-tests":       <*matchers.ContainSubstringMatcher | 0xc0028979e0>: {Substr: "EOF", Args: nil}[0m
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   [38;5;9mIn [1m[It][0m[38;5;9m at: [1m/workspace/gloo/test/e2e/local_ratelimit_test.go:161[0m [38;5;243m@ 12/02/24 23:28:17.819[0m
Step #1 - "run-tests": Step #8 - "run-e2e-tests": 
Step #1 - "run-tests": Step #8 - "run-e2e-tests":   [38;5;9mFull Stack Trace[0m
Step #1 - "run-tests": Step #8 - "run-e2e-tests":     github.com/solo-io/gloo/test/e2e_test.init.func29.6.2()
Step #1 - "run-tests": Step #8 - "run-e2e-tests":     	/workspace/gloo/test/e2e/local_ratelimit_test.go:161 +0x9d
```

**Review by nfuden — APPROVED** [review]

File: .github/workflows/nightly-tests.yaml [hunk 2]
```diff
@@ -4,24 +4,17 @@ env:
...
+  # schedule:
+  # -
   workflow_dispatch:
     inputs:
       branch:
         description: "The branch to run tests against"
         type: choice
         options:
           - main
+          - v1.18.x
```
Comment (sheidkamp) [inline .github/workflows/nightly-tests.yaml:17]: If we are going to remove the jobs that run for all of the options (everywhere `inputs.branch ==` got deleted) except for `v1.18.x`, should we make it the only option?

**Review by sheidkamp — COMMENTED** [review]
One suggestion

**Review by sheidkamp — APPROVED** [review]

**Review by jenshu — APPROVED** [review]

**Pull request merged**